tango_simlib/tests/test_yaml_builder.py

```diff
@@ -0,0 +1,278 @@
+#########################################################################################
+# Copyright 2020 SKA South Africa (http://ska.ac.za/)                                   #
+#                                                                                       #
+# BSD license - see LICENSE.txt for details                                             #
+#########################################################################################
+"""Tests for YAML builder"""
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+from argparse import Namespace
+from pathlib import Path
+
+import yaml
+import tango
+
+from tango_simlib.tango_yaml_tools.main import _build_yaml
+
+CONF_FILE_PATH = Path.joinpath(Path(__file__).parent, "config_files")
+
+
+def validate_basic_structure(yaml_dict):
+    """Do a basic check of the structure
+    """
+    assert len(yaml_dict) == 1
+    assert "meta" in yaml_dict[0]
+    assert "class" in yaml_dict[0]
+    assert "properties" in yaml_dict[0]["meta"]
+    assert "attributes" in yaml_dict[0]["meta"]
+
```

Comment:
Aren't going to check for the commands key here?

Reply:
I should 😄